api/src/main/java/io/opentelemetry/common/Main.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.common;
+
+import java.util.List;
+
+@SuppressWarnings({
+  "BadImport",
+  "UnusedVariable",
+  "PrivateConstructorForUtilityClass",
+  "MultipleTopLevelClasses",
+  "JavadocMethod"
+})
+public class Main {
+
+  public static void main(String[] args) {
+    CleanAttributes attributes =
+        CleanAttributes.newBuilder()
+            .setAttribute("string", "I'm a String")
+            .setAttribute("boolean", true)
+            .setAttribute("double", 33.444d)
+            .setAttribute("long", 34333L)
+            .setAttribute("stringArray", "one", "two", "three")
+            .setAttribute("booleanArray", true, false, true)
+            .setAttribute("longArray", 33L, 55L, 99L)
+            .setAttribute("doubleArray", 123.33, 6655.33, 339393.33, 3434.33)
+            .build();
+
+    process(attributes);
+  }
+
+  public static void process(final CleanReadableAttributes attributes) {
+    System.out.println("attributes = " + attributes);
+    attributes.forEach(
+        new CleanReadableAttributes.AttributeConsumer() {
+          @Override
+          public void consume(String key, AttributeValue.Type type, Object value) {
+            switch (type) {
+              case STRING:
+                String stringValue = attributes.getStringValue(value);
```

Comment:
This could just be `(String) value`, there doesn't seem to be much value to delegate the cast.

Reply:
yeah, I have gone back and forth on this one. I agree with you, but I also like having the option for the non-obvious ones (like the "array" values that are actually Lists). You can definitely just have the consumer do the casts manually, for sure.

Reply:
Can you clarify "non-obvious"? I guess just that it looks a little gross (which I agree with but not enough to make the indirect API I think)

`List<String> stringArrayValue = (List<String>) attributes.getStringArrayValue(value);`

That being said, I think requiring casts does have the downside of requiring the suppresswarnings "unchecked" too if those are enabled. I guess if we provide a helper whose purpose is a cast, we could make it static instead of going through the attributes instance.

Reply:
`getStringArrayValue` implies `String[]` not `List<String>`

Reply:
How about `String AttributeValue.Type.STRING.cast(Object value)` and similar for other types? That will clean up the API of `Attributes`

Reply:
Would the signature be `<T> T cast(Object value)`? I guess it needs to be same signature on all the enum values. This seems ok to me though as it's pretty readable.

Reply:
It doesn't need to be the same:
```
    STRING {
      public String cast(Object value){
        return (String) value;
      }
    },
    BOOLEAN{
      public boolean cast(Object value){
        return (boolean) value;
      }
    },
```

Reply:
Ah yeah haven't used that pattern much before - great trick for this case!

Reply:
excellent. something like this has been tickling the back of my brain, but I couldn't quite put my finger on it. I will try this out and add this API to this PR to see how it feels/looks.

Reply:
hmm. I'm not sure how you get a handle to call those methods. They don't appear to be visible to be called anywhere. Java doesn't let you add static members to enum instances, either, unfortunately.

Reply:
yeah, you can't call methods like that on enum values. See here for the JEP to make it possible:  http://openjdk.java.net/jeps/301

Reply:
I moved the methods to do the casts into a new AttributeType class, which just moves the issue to a different place. I haven't been able to figure out a way to make it so you can only call the right one, yet.

Reply:
I also added an 8-way typed consumer interface, just to see how that worked out. With the requirement for a consumer to do the casts with the single method, 8 methods doesn't look too bad. But, also doesn't have the advantage of being a SMI for implementation via a lambda.